### ISSUE TITLE
Only functional cheat with weapon-config

### ISSUE THREAD

**Issue by CeloTL** [oscar-broman/samp-weapon-config#152]
I found another flaw in Weapon-Config, unfortunately this time I couldn't find the solution. It is a cheat damage that kills everyone nearby.
Video: https://www.youtube.com/watch?v=-v3CYkgk2EY&feature=youtu.be
Cheat for testing: https://www.mediafire.com/file/vgwayii2ix13opk/DamagerTL.rar/file
/damager and keep pressing R

**Comment by ADRFranklin** [comment]
I am aware of this and know a solution to it. Thanks for the report.

**Comment by CeloTL** [comment]
Thank you! If you need to do some tests, I can help.

**Comment by CeloTL** [comment]
Good evening!
Any progress? My server and many others who enjoy and use their work are suffering from this flaw that spoils people's gameplay.

**Comment by ADRFranklin** [comment]
I haven't had a chance to fix it, I have a lot of other projects to work on right now, and I am just now finding time to look at the issue, though I probably won't have a solution ready for a while.

**Comment by CeloTL** [comment]
Hello. Any news of repair? Unfortunately mine and many other servers have been suffering from this failure.
The job is great, what spoils is the people taking advantage of the failures.

**Comment by ADRFranklin** [comment]
Sorry, I have been busy with work and projects like sampctl recently, I need to get some minor stuff out the way over the next few days, but I am fairly sure I have time to work on a fix for it.

**Comment by xunder-matth** [comment]
@CeloTL you can implement a quick fix using "OnRejectedHit" callback with reason HIT_MULTIPLE_PLAYERS. It's not the most brilliant way to fix this cheat but it works fine for me and I didn't receive false positives yet...

**Comment by ADRFranklin** [comment]
@xunder-matth Would you be willing to share some code or make a PR for now? At least until this problem can be fully addressed.

**Comment by xunder-matth** [comment]
@ADRFranklin I already sent you debug on Discord. 
https://i.imgur.com/rnVZ8LJ.png - Where Xunder_Testic is cheater and __xunder is not.
Here is how I "fixed" it:
```
public OnRejectedHit(playerid,hit[E_REJECTED_HIT]) {
	if (hit[e_Reason] == HIT_MULTIPLE_PLAYERS) {
		new hits = hit[e_Info1],
			buffer[96];

	    format(buffer, 128, "~r~AC:~y~ %s (%d)~w~ possible~r~ Damager Hack (%d players)", Player_Name[playerid], playerid, hits);
		SendAltStaffMessage(buffer);
	}
	return 1;
}
```
For some reason, "Damager hack" will always hit **6** players with one bullet.

**Comment by CeloTL** [comment]
I will add it to the server and do the tests, thank you for your attention and effort.

**Comment by AliLogic** [comment]
[This comment](https://github.com/oscar-broman/samp-weapon-config/issues/152#issuecomment-663905682) shouldn't actually work because the `OnRejectedHit` is called when the hit is rejected as the name says, which means hit is ignored and no dmg is given.  The temporary fix would be to update to 0.3.7 R3 or 0.3.7 R4 by forcing players to use it (you should be on the latest version regardless) until I possibly look into fixing it.

**Comment by xunder-matth** [comment]
@AliLogic that's why I put fix in quotes. This will not prevent the cheater from giving damage but it will output a warning message after some time. Anyway, there is a way to bypass this "fix" and as I said before it's not perfect...

> 
> 
> The temporary fix would be to update to 0.3.7 R3 or 0.3.7 R4 by forcing players to use it (you should be on the latest version regardless) since SAMPFUNCS doesn't work on it until I possibly look into fixing it.

Unfortunately, that method is easily bypassable with version fakers...

**Comment by CeloTL** [comment]
> @AliLogic that's why I put fix in quotes. This will not prevent the cheater from giving damage but it will output a warning message after some time. Anyway, there is a way to bypass this "fix" and as I said before it's not perfect...
> 
> > The temporary fix would be to update to 0.3.7 R3 or 0.3.7 R4 by forcing players to use it (you should be on the latest version regardless) since SAMPFUNCS doesn't work on it until I possibly look into fixing it.
> 
> Unfortunately, that method is easily bypassable with version fakers...


It is true, I have already tried such a maneuver.
